### PR TITLE
Name the dataloader shuffling seed

### PR DESCRIPTION
**Pull request by Kh4L**
Signed-off-by: Serge Panev <spanev@nvidia.com>

#### Why we need this PR?
To give a meaning to the value and to avoid duplicate declarations of the value.

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *[ Explain solution of the problem, a new feature added here. ]*
 - Affected modules and functionalities:
     *[ Describe here what was changed, added, removed. ]*
 - Key points relevant for the review:
     *[ Describe here what is the most important part that reviewers should focus on. ]*
 - Validation and testing:
     *[ Describe here if and how this PR is tested. ]*
 - Documentation (including examples):
     *[ Describe here if documentation and examples were updated. ]*

### CODE REVIEW

**Comment by Kh4L** [comment]
!build

File: include/dali/core/common.h [hunk 1]
```diff
@@ -259,4 +259,6 @@ DLL_PUBLIC std::vector<std::string> string_split(const std::string &s, const cha
   if (1) \
   std::cerr << __FILE__ << ":" << __LINE__ << ": "
 
+#define DALI_DATALOADER_SEED 524287
```
Comment (JanuszL) [inline include/dali/core/common.h]: Why not a constexpr?

Comment (Kh4L) [reply include/dali/core/common.h]: I don't know, I hesitated between define and constexpr

Comment (Kh4L) [reply include/dali/core/common.h]: I will change to constexpr

Comment (JanuszL) [reply include/dali/core/common.h]: I don't have any strong opinion either. @jantonguirao - please share your thought.

Comment (Kh4L) [reply include/dali/core/common.h]: Ok let's wait for his opinion

Comment (jantonguirao) [reply include/dali/core/common.h]: A constexpr is better in my opinion

Comment (Kh4L) [reply include/dali/core/common.h]: Done

**Comment by Kh4L** [comment]
!build

**Pull request merged**